react/features/transcription/middleware.js

```diff
@@ -0,0 +1,126 @@
+
+/**
+* Time after which the rendered subtitles will be removed.
+*/
+const REMOVE_AFTER_MS = 3000;
+
+/**
+ * Middleware that catches actions related to transcript messages
+ * to be rendered in {@link TranscriptionSubtitles }
+ *
+ * @param {Store} store - Redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+
+    switch (action.type) {
+    case ENDPOINT_MESSAGE_RECEIVED:
+        return _endpointMessageReceived(store, next, action);
+    }
+
+    return next(action);
+});
+
+/**
+ * Notifies the feature transcription that the action
+ * {@code ENDPOINT_MESSAGE_RECEIVED} is being dispatched within a specific redux
+ * store.
+ *
+ * @param {Store} store - The redux store in which the specified {@code action}
+ * is being dispatched.
+ * @param {Dispatch} next - The redux {@code dispatch} function to
+ * dispatch the specified {@code action} to the specified {@code store}.
+ * @param {Action} action - The redux action {@code ENDPOINT_MESSAGE_RECEIVED}
+ * which is being dispatched in the specified {@code store}.
+ * @private
+ * @returns {Object} The value returned by {@code next(action)}.
+ */
+function _endpointMessageReceived({ dispatch, getState }, next, action) {
+    const payload = action.payload;
+
+    try {
+
+        // Let's first check if the given object has the correct
+        // topic in the payload, which identifies it as a json message send
+        // from Jigasi with speech-to-to-text results
+        // if (m['jitsi-meet-muc-msg-topic'] === TRANSCRIPTION_RESULT_TOPIC
+        //         && (payload = m.payload))
+        // TODO: To be replaced by if(payload['topic']==='transcription-result')
+        // after jigasi changes.
+        if (payload.transcript) {
+            // Extract the useful data from the payload of the JSON message
+            const isInterim = payload.is_interim;
+            const participantName = payload.participant.name;
+            const stability = payload.stability;
+            const text = payload.transcript[0].text;
+            const transcriptMessageID = payload.message_id;
+
+            // If this is the first result with the unique message ID,
+            // we add it to the state along with the name of the participant
+            // who said given text
+            if (!getState()['features/transcription']
+                .transcriptMessages.get(transcriptMessageID)) {
+                dispatch(addTranscriptMessage(transcriptMessageID,
+                    participantName));
+            }
+
+            // If this is final result, update the state as a final result
+            // and start a count down to remove the subtitle from the state
+            if (!isInterim) {
+                const { transcriptMessages }
+                    = getState()['features/transcription'];
+                const newTranscriptMessage
+                    = transcriptMessages.get(transcriptMessageID);
```

Comment:
But, I can move this fetch of `transcriptMessages` out of this if block, which won't decrease the no. of calls as such ( as they are in if elseif else block), but will just be neater.